CorpusCallosum/data/read_write.py

```diff
@@ -0,0 +1,279 @@
+# Copyright 2025 AI in Medical Imaging, German Center for Neurodegenerative Diseases(DZNE), Bonn
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import multiprocessing
+
+import nibabel as nib
+import numpy as np
+
+import FastSurferCNN.utils.logging as logging
+
+logger = logging.get_logger(__name__)
+
+
+def run_in_background(function, debug=False, *args, **kwargs):
+    """Run a function in the background using multiprocessing.
+
+    This function executes the given function either in a separate process (normal mode)
+    or in the current process (debug mode). In debug mode, the function is executed
+    synchronously for easier debugging.
+
+    Args:
+        function: The function to execute
+        debug (bool): If True, run synchronously in current process
+        args: Positional arguments to pass to the function
+        kwargs: Keyword arguments to pass to the function
+
+    Returns:
+        multiprocessing.Process or None: Process object if running in background,
+        None if in debug mode
+    """
+    if debug:
+        function(*args, **kwargs)
+        process = None
+    else:
+        process = multiprocessing.Process(target=function, args=args, kwargs=kwargs)
+        process.start()
+    return process
+
+
+
+def get_centroids_from_nib(seg_img: nib.Nifti1Image, label_ids: list[int] | None = None) -> dict[int, np.ndarray]:
```

Comment:
I dont like the name of this function. `get_` is not descriptive enough for me, `compute_` might be better. Also, given that `seg_img` is directly converted to data and vox2ras, I think it might better to just have those two be the arguments directly and then also the "from_nib" is less obtuse. now this is an io and compute function at the same time with limited reason.
Finally the return type in the annotation is incorrect, it is either that or the tuple. Good - typing-compatible Fix (also needs from typing import overload`:
```suggestion
@overload
def get_centroids_from_nib(seg_img: nib.Nifti1Image, label_ids: None = None) -> dict[int, np.ndarray]:
    ...

@overload
def get_centroids_from_nib(seg_img: nib.Nifti1Image, label_ids: list[int] | None = None) -> tuple[dict[int, np.ndarray], list[int]]:
    ...

def get_centroids_from_nib(seg_img: nib.Nifti1Image, label_ids: list[int]) -> dict[int, np.ndarray]:
    ...
```